lib/types/uuid.js

```diff
@@ -2,126 +2,150 @@
 
 const crypto = require("crypto");
 const utils = require("../utils");
+const rust = require("../../index");
 
 /** @module types */
 
 /**
- * Creates a new instance of Uuid based on a Buffer
  * @class
  * @classdesc Represents an immutable universally unique identifier (UUID). A UUID represents a 128-bit value.
- * @param {Buffer} buffer The 16-length buffer.
- * @constructor
  */
-function Uuid(buffer) {
-    if (!buffer || buffer.length !== 16) {
-        throw new Error("You must provide a buffer containing 16 bytes");
+class Uuid {
+    /**
+     * Creates a new instance of Uuid based on a Buffer
+     * @param {Buffer} buffer The 16-length buffer.
+     * @constructor
+     */
+    constructor(buffer) {
+        if (!buffer || buffer.length !== 16) {
+            throw new Error("You must provide a buffer containing 16 bytes");
+        }
+        this.internal = rust.UuidWrapper.new(buffer);
     }
-    this.buffer = buffer;
-}
 
-/**
- * Parses a string representation of a Uuid
- * @param {String} value
- * @returns {Uuid}
- */
-Uuid.fromString = function (value) {
-    //36 chars: 32 + 4 hyphens
-    if (typeof value !== "string" || value.length !== 36) {
-        throw new Error(
-            "Invalid string representation of Uuid, it should be in the 00000000-0000-0000-0000-000000000000",
+    /**
+     * Parses a string representation of a Uuid
+     * @param {String} value
+     * @returns {Uuid}
+     */
+    static fromString(value) {
+        // 36 chars: 32 + 4 hyphens
+        if (typeof value !== "string" || value.length !== 36) {
+            throw new Error(
+                "Invalid string representation of Uuid, it should be in the 00000000-0000-0000-0000-000000000000 format",
+            );
+        }
+        return new Uuid(
+            utils.allocBufferFromString(value.replace(/-/g, ""), "hex"),
         );
     }
-    return new Uuid(
-        utils.allocBufferFromString(value.replace(/-/g, ""), "hex"),
-    );
-};
 
-/**
- * Creates a new random (version 4) Uuid.
- * @param {function} [callback] Optional callback to be invoked with the error as first parameter and the created Uuid as
- * second parameter.
- * @returns {Uuid}
- */
-Uuid.random = function (callback) {
-    if (callback) {
-        getRandomBytes(function (err, buffer) {
-            if (err) {
-                return callback(err);
+    /**
+     * Creates a new random (version 4) Uuid.
+     * @param {function} [callback] Optional callback to be invoked with the error as first parameter and the created Uuid as
+     * second parameter.
+     * @returns {Uuid}
+     */
+    static random(callback) {
+        if (callback) {
+            getRandomBytes(function (err, buffer) {
+                if (err) {
+                    return callback(err);
+                }
+                return callback(null, createUuidFromBuffer(buffer));
+            });
+        } else {
+            const buffer = getRandomBytes();
```

Comment:
Simple helper function with two call. Is it required to be a helper function then?

Reply:
We pass a callback to it so I think it improves readability a little bit.

Reply:
Maybe consider then moving the helper function to the body of this function: like here?
https://github.com/scylladb-zpp-2024-javascript-driver/scylladb-javascript-driver/blob/03317eeda2165e200cb006cd772ce8178ffb0b5d/lib/types/duration.js#L107-L125